x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/ShapeFieldMapper.java

```diff
@@ -76,13 +76,14 @@ public Builder newBuilder(String name, Map<String, Object> params) {
         }
     }
 
-    public static final class ShapeFieldType extends AbstractShapeGeometryFieldType<Geometry, Geometry>
+    public static final class ShapeFieldType extends AbstractShapeGeometryFieldType
         implements ShapeQueryable {
 
         private final ShapeQueryProcessor queryProcessor;
 
-        public ShapeFieldType(String name, boolean indexed, boolean stored, boolean hasDocValues, Map<String, String> meta) {
-            super(name, indexed, stored, hasDocValues, false, meta);
+        public ShapeFieldType(String name, boolean indexed, boolean stored, boolean hasDocValues,
+                              Parser<Geometry> parser, Map<String, String> meta) {
```

Comment:
Note to myself: This constructor is public because it used by a test. It should probably be private but let address it in a follow up PR.